services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -0,0 +1,147 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import logging
+
+import duckdb
+from libcommon.constants import PROCESSING_STEP_SPLIT_DUCKDB_INDEX_VERSION
+from libcommon.exceptions import (
+    NoIndexableColumnsError,
+    ParquetResponseEmptyError,
+    PreviousStepFormatError,
+    UnsupportedIndexableColumnsError,
+)
+from libcommon.processing_graph import ProcessingStep
+from libcommon.storage import StrPath
+from libcommon.utils import JobInfo
+from libcommon.viewer_utils.index_utils import create_index_dir_split
+
+from worker.config import AppConfig
+from worker.job_runners.split.split_job_runner import SplitJobRunner
+from worker.utils import (
+    CompleteJobResult,
+    IndexRowsResponse,
+    get_previous_step_or_raise,
+)
+
+STRING_FEATURE_DTYPE = "string"
+VALUE_FEATURE_TYPE = "Value"
+DUCKDB_DEFAULT_DB_NAME = "index.db"
```

Comment:
nit: here and in other places where you mention `db_name`, maybe replace with `index_filename`, which is more meaningful (at least for me)

Reply:
I changed it to DUCKDB_DEFAULT_INDEX_FILENAME